File: src/main/java/io/bindernews/thegrackle/cards/InFlightService.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.actions.common.ObtainPotionAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.potions.RegenPotion;
import com.megacrit.cardcrawl.powers.RegenPower;
import io.bindernews.bnsts.CardVariables;
import lombok.val;

import static io.bindernews.thegrackle.helper.ModInterop.iop;

public class InFlightService extends BaseCard {
    public static final CardConfig C =
            new CardConfig("InFlightService", CardType.SKILL, CardRarity.RARE, CardTarget.SELF);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(2, 1);
        c.magic(5, -1);
        c.exhaust(true, true);
    });

    public InFlightService() {
        super(C, VARS);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        if (p instanceof AbstractPlayer) {
            addToBot(new ObtainPotionAction(new RegenPotion()));
        } else {
            val power = iop().createPower(RegenPower.POWER_ID, p, magicNumber);
            addToBot(new ApplyPowerAction(p, p, power, magicNumber));
        }
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/InFlightService.kt
package io.bindernews.thegrackle.cards

import basemod.cardmods.ExhaustMod
import com.megacrit.cardcrawl.actions.common.ObtainPotionAction
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.potions.RegenPotion
import com.megacrit.cardcrawl.powers.RegenPower
import io.bindernews.bnsts.CardVariables
import io.bindernews.thegrackle.helper.ModInterop.iop

class InFlightService : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature) {
        if (p is AbstractPlayer) {
            addToBot(ObtainPotionAction(RegenPotion()))
        } else {
            addToBot(iop().actionApplyPower(p, p, RegenPower.POWER_ID, magicNumber))
        }
    }

    companion object {
        @JvmStatic val C = CardConfig("InFlightService", CardType.SKILL, CardRarity.RARE, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(2, 1)
            c.magic(5, -1)
            c.addModifier(ExhaustMod())
        }
    }
}